feat(dli): add new datasource to get list of sql jobs (#4684)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/dli_sql_jobs.md

---
subcategory: "Data Lake Insight (DLI)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_dli_sql_jobs"
description: |-
  Use this data source to get the list of the DLI sql jobs.
---

# huaweicloud_dli_sql_jobs

> Use this data source to get the list of the DLI sql jobs.

Reviewer: ```suggestion
Use this data source to get the list of the DLI SQL jobs.
```

Author: done

## Example Usage

```hcl
variable "job_id" {}

data "huaweicloud_dli_sql_jobs" "test" {
  job_id = var.job_id
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the resource.
  If omitted, the provider-level region will be used.

* `job_id` - (Optional, String) Specifies the ID of the job to be queried.

> * `type` - (Optional, String) Specifies the type of the job to be queried.

Reviewer: ```suggestion
* `type` - (Optional, String) Specifies the type of the jobs to be queried.
```

Author: done

* `status` - (Optional, String) Specifies the status of the job to be queried.
  The valid values are **FINISHED**, **FAILED** and **CANCELED**.

> * `queue_name` - (Optional, String) Specifies the queue name which this job to be submitted belongs.

Reviewer: ```suggestion
* `queue_name` - (Optional, String) Specifies the queue name which the jobs to be submitted belong.
```

Author: done

* `start_time` - (Optional, Int) Specifies the time when a job is start to be queried.
  Is a timestamp in milliseconds, e.g. `1713182789931`.

* `end_time` - (Optional, Int) Specifies the time when a job is end to be queried.
  Is a timestamp in milliseconds, e.g. `1713182790346`.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The data source ID.

* `jobs` - All jobs that match the filter parameters.

  The [jobs](#jobs_struct) structure is documented below.

<a name="jobs_struct"></a>
The `jobs` block supports:

* `id` - The ID of job.

* `type` - The type of job.

* `status` - The status of the job.

* `queue_name` - The queue name which this job to be submitted belongs.

* `owner` - The user who submits the job.

* `database_name` - The database name where the table that records its operations is located.

* `sql` - The SQL statement is executed by the job.

> * `start_time` - The time when a job is start.
> 
> * `end_time` - The time when a job is end.

Reviewer: ```suggestion
* `start_time` - The time when a job is start, in RFC 3339 format.

* `end_time` - The time when a job is end, in RFC 3339 format.
```

Author: done

* `duration` - The job running duration (unit: millisecond).

> * `tags` - The label of the job.

Reviewer: ```suggestion
* `tags` - The key/value pairs to associate with the job.
```

Author: done
